### PR TITLE
Add k8s ingress configs with and without nginx

### PR DESCRIPTION
**Pull request by rmars**
* add a third service (api) that calls hello
* refactor l5d_headers() into its own file
* stop exposing the hello service in our other configs

Todos:

- [x]  stop hardcoding linkerd's IP in nginx.conf
- [x] use "real" images insead of helloworld:test7 and helloworld:nginx3

### CODE REVIEW

File: k8s-daemonset/k8s-ingress/linkerd.yml [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    - protocol: http
+      label: ingress
+      baseDtab: |
+        /world/hello/www => /#/io.l5d.k8s/default/http/hello ;
+        /world/hello/api => /#/io.l5d.k8s/default/http/api ;
+        /host            => /$/io.buoyant.http.domainToPath ;
+        /http/*/*        => /host ;
+      servers:
+      - port: 80
+        ip: 0.0.0.0
```
Comment (klingerf) [inline k8s-daemonset/k8s-ingress/linkerd.yml]: Instead of serving this router on port 80, I think it would be more conventional to have it serve on a higher port (4142?), and then map that port to port 80 in the l5d service object.

File: k8s-daemonset/k8s-ingress/nginx.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    app: nginx
+  template:
+    metadata:
+      name: nginx
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - name: nginx
+        image: buoyantio/helloworld:nginx4
```
Comment (klingerf) [inline k8s-daemonset/k8s-ingress/nginx.yml]: Maybe we should call this image `buoyant/nginx-more`?  Would also be good to check the Dockerfile for that image into the `docker` top-level directory in this repo.

File: k8s-daemonset/k8s-ingress/nginx/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,10 @@
+FROM nginx
```
Comment (klingerf) [inline k8s-daemonset/k8s-ingress/nginx/Dockerfile]: Per comment above, I'd moved all of the files needed to build to docker images into a `docker/nginx-headers-more` dir or something similar.

File: k8s-daemonset/k8s/hello-world.yml [hunk 4]
```diff
@@ -50,7 +50,7 @@ metadata:
 spec:
   selector:
     app: hello
-  type: LoadBalancer
+  type: ClusterIP
```
Comment (klingerf) [inline k8s-daemonset/k8s/hello-world.yml]: I think you can revert this change, since you're not using this config file for the ingress example.

Comment (rmars) [inline k8s-daemonset/k8s/hello-world.yml]: Alex pointed out that we shouldn't be exposing the hello service regardless (in the other example I think we only access it via linkerd) - he suggested removing it

Comment (klingerf) [inline k8s-daemonset/k8s/hello-world.yml]: Ah, ok, makes sense -- in that case can you make the same change in `hello-world-1_4.yml`?

File: k8s-daemonset/k8s/api.yml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    app: api
+  template:
+    metadata:
+      labels:
+        app: api
+    spec:
+      dnsPolicy: ClusterFirst
+      containers:
+      - name: service
+        image: buoyantio/helloworld:test7
```
Comment (klingerf) [inline k8s-daemonset/k8s/api.yml]: Cool, let's push your changes to the app as 0.0.4, and change all of the references (in this file and in the hello-world*.yml files) to use the 0.0.4 image.

File: k8s-daemonset/k8s/linkerd-ingress.yml [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+      labels:
+        app: l5d
+    spec:
+      volumes:
+      - name: l5d-config
+        configMap:
+          name: "l5d-config"
+      containers:
+      - name: l5d
+        image: buoyantio/linkerd:0.8.2
```
Comment (klingerf) [inline k8s-daemonset/k8s/linkerd-ingress.yml]: Can you bump this to 0.8.3?

File: docker/nginx/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,10 @@
+FROM nginx
```
Comment (klingerf) [inline docker/nginx/Dockerfile]: Let's pin this explicitly to the current nginx version:

```
FROM nginx:1.11.5
```

And then we should tag the image that you build from this file as buoyantio/nginx:1.11.5.

File: k8s-daemonset/k8s/nginx.yml [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    app: nginx
+  template:
+    metadata:
+      name: nginx
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - name: nginx
+        image: buoyantio/nginx:latest
```
Comment (klingerf) [inline k8s-daemonset/k8s/nginx.yml]: And let's use `buoyantio/nginx:1.11.5` here as well.

File: k8s-daemonset/k8s/api.yml (hunk 5 above)
Comment (klingerf) [inline k8s-daemonset/k8s/api.yml]: Good question! My preference is to keep all of the configs pinned at the same version.
